src/Language/PureScript/CoreFn/Optimizer.hs

```diff
@@ -48,9 +51,17 @@ closedRecordFields (TypeApp _ (TypeConstructor _ C.Record) row) =
 closedRecordFields _ = Nothing
 
 -- | See https://github.com/purescript/purescript/issues/3157
-optimizeUnusedPartialFn :: Expr a -> Expr a
+optimizeUnusedPartialFn :: Expr Ann -> Expr Ann
 optimizeUnusedPartialFn (Let _
   [NonRec _ UnusedIdent _]
   (App _ (App _ (Var _ (Qualified _ UnusedIdent)) _) originalCoreFn)) =
   originalCoreFn
 optimizeUnusedPartialFn e = e
+
+-- | TODO: Fixup the annotations here.
```

Comment:
The annotation here seems mostly correct, as this currently points to the function application as a whole.

Reply:
Agree, I think you're doing the right thing with annotations.